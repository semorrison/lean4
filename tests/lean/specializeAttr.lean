@[specialize f g h]
def f1 (f : Nat → Nat) (g : Nat → Nat) (x : Nat) :=
  f (f (g x))

@[specialize f 1 g]
def f2 (f : Nat → Nat) (g : Nat → Nat) (x : Nat) :=
  f (f (g x))

@[specialize 1 f g]
def f3 (f : Nat → Nat) (g : Nat → Nat) (x : Nat) :=
  f (f (g x))

@[specialize 0 g]
def f4 (f : Nat → Nat) (g : Nat → Nat) (x : Nat) :=
  f (f (g x))

@[specialize 10]
def f5 (f : Nat → Nat) (g : Nat → Nat) (x : Nat) :=
  f (f (g x))

@[specialize ff]
def f6 (f : Nat → Nat) (g : Nat → Nat) (x : Nat) :=
  f (f (g x))
