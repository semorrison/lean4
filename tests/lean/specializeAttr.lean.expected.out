specializeAttr.lean:1:17-1:18: error: invalid specialization argument name `h`, `f1` does have an argument with this name
specializeAttr.lean:5:15-5:16: error: invalid specialization argument index, `f` has already been specified as a specialization candidate
specializeAttr.lean:9:15-9:16: error: invalid specialization argument name `f`, it has already been specified as a specialization candidate
specializeAttr.lean:13:13-13:14: error: invalid specialization argument index, index must be greater than 0
specializeAttr.lean:17:13-17:15: error: invalid argument index, `f5` has #3 arguments
specializeAttr.lean:21:13-21:15: error: invalid specialization argument name `ff`, `f6` does have an argument with this name
